File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    days = deque()
    for i in range(len(progresses)):
        count = 0
        while progresses[i] < 100:
            progresses[i] += speeds[i]
            count += 1
        days.append(count)
    
    result = []
    while days:
        day = days.popleft()
        a = 1
        while days and day >= days[0]:
            a += 1
            days.popleft()
        
        result.append(a)
        
    return result

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> days = new ArrayList<>();
        
        for (int i = 0; i < speeds.length; i++) {
            days.add((int) Math.ceil((100.0 - progresses[i]) / speeds[i]));
        }
        
        int cnt = 0;
        int curDay = days.get(0);
        List<Integer> answer = new ArrayList<>();
        for (int i = 1; i < days.size(); i++) {
            cnt += 1;
            if (curDay < days.get(i)) {
                answer.add(cnt);
                cnt = 0;
                curDay = days.get(i);
            }
        }
        answer.add(cnt+1);
        return answer.stream().mapToInt(i -> i).toArray();
    }
}